src/src/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -790,7 +785,7 @@ protected TokenTaskResult doInBackground(String... urlItems) {
                 // Record account in the AccountManager service
                 try {
                     setAccount(result);
-                } catch (Exception exc) {
+                } catch (GeneralSecurityException | IOException exc) {
```

Comment:
Here is our current way to log errors
```
    public void error(String tag, String message, String additionalMessage, ADALError errorCode) {
        if (mAndroidLogEnabled) {
            Log.e(tag, getLogMessage(message, additionalMessage, errorCode));
        }

        logCommon(tag, message, additionalMessage, LogLevel.Error, errorCode);
    }
```
if mAndroidLogEnabled is set to be false, Log.e won't be called. We won't know the exact exception thrown. i.e For the following decrption of appIdList, we're catching the GeneralSecurityException and IOException, and we log like "appUIDList failed to decrypt" and then set appIdList to be empty. This is the broker flow, and when we next time trying to find a matching cache item from broker, this could cause failure of uid check in broker. This is basically why i also just want us the log the exception name and exception message in the log message itself. 

I also agree we should improve Logger, we already passed in the exception today, Logger should be improved to handle this.